File: app/javascript/channels/chat_channel.js
/* eslint-disable no-console */
import consumer from './consumer';

consumer.subscriptions.create('ChatChannel', {
  connected() {
    console.log('ActionCable connected.');
    this.speak();
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    console.log('ActionCable disconnected.');
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    console.log('ActionCable received:')
    console.log(data);
    // Called when there's incoming data on the websocket for this channel
  },

  speak() {
    console.log('ActionCable speaks.');
    return this.perform('speak', {});
  },
});

File: app/channels/chat_channel.rb
# frozen_string_literal: true

class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from 'room'
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak
    ActionCable.server.broadcast('room', {'key' => 'value'})
  end
end
